lib/generate/orderModules.js

```diff
@@ -0,0 +1,65 @@
+const logger = require('../utils/logger');
+
+/**
+ * Given a list of modules and their dependencies, return list in a consistent
+ * order, whilst still ensuring dependencies are loaded before their dependant
+ * modules.
+ */
+module.exports = (modules, dependencies) => {
+    // First we set up a map with the existing dependencies. These are obtained
+    // by listening to requirejs onResourceLoad event, so are in a working but
+    // non-deterministic order, based on browser response times.
+    const dependencyMap = new Map(Object.entries(dependencies));
+
+    // We then set up a new `orderedModules` object and transfer `dependencyMap`
+    // to it, adding modules in a consistent order. Objects are moved out of
+    // dependency map, so we loop until it has 0 length, bailing if it seems
+    // like we're unable to proceed.
+    const orderedModules = {};
+    let level = 0;
+    while (dependencyMap.size) {
+        if (level++ > 999) {
+            logger.warn(
+                'Unable to determine consistent module order. Falling back to known-working order (which might be different from the last time this ran).'
+            );
+            logger.debug(modules, dependencies);
```

Comment:
Including `dependencyMap` in this output makes debugging _why_ this happens easier.
```suggestion
            logger.debug(modules, dependencies, dependencyMap);
```